packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -673,9 +673,16 @@ class AndroidDevice extends Device {
   }
 
   @override
-  FutureOr<DeviceLogReader> getLogReader({ AndroidApk app }) async {
+  FutureOr<DeviceLogReader> getLogReader({
+    AndroidApk app,
+    bool includePastLogs = false,
+  }) async {
     // The Android log reader isn't app-specific.
-    return _logReader ??= await AdbLogReader.createLogReader(this, globals.processManager);
+    return _logReader ??= await AdbLogReader.createLogReader(
```

Comment:
this now needs two cached `_logReaders` to account for  `includePastLogs`

Reply:
Good catch, thanks!